src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -866,24 +928,30 @@ class SegmentCleaner : public SegmentProvider {
   public:
     GCProcess(SegmentCleaner &cleaner) : cleaner(cleaner) {}
 
+    bool is_stopped() const {
```

Comment:
This should be private, and I think it should be called is_stopping().  A user would reasonably assume that it's ok to destruct the SegmentCleaner object once is_stopped() returns true, but here stop() and the background gc process won't necessarily have actually resolved yet.

Reply:
Yeah, that's true.

Reply:
see https://github.com/ceph/ceph/pull/46158/commits/213f1b37bea8323216c34f087be63c901bbc20a9